patcher.sh

```diff
@@ -0,0 +1,330 @@
+
+		# Move APK
+		mv "$tempApkDir/AM2RWrapper-aligned-debugSigned.apk" "$SCRIPT_DIR/AndroidM2R_"$VERSION"-signed.apk"
+
+	else
+		>&2 echo "Invalid OS \"$OSCHOICE\"! Cannot patch anything!"
+		return 1
+	fi
+
+	echo ""
+	echo "The operation was completed successfully. See you next mission!"
+	return 0
+}
+
+# ----
+# Main function starts here
+# ---
+main ()
+{
+	echo "-------------------------------------------"
+	echo ""
+	echo "AM2R ${VERSION} Shell Autopatching Utility"
+	echo "Scripted by Lojemiru and Miepee"
+	echo ""
+	echo "-------------------------------------------"
+
+	if (( $# <= 0 )); then
+		APPIMAGE=true
+		AM2RZIP="$SCRIPT_DIR/AM2R_11.zip"
+		local input=""
+		echo "Select your patch type:"
+		echo ""
+		echo "1 - Linux"
+		echo "2 - Android"
+		echo ""
+		echo "Awaiting input:"
+		read -n1 input
+		echo ""
+		if [[ "${input}" = "1" ]]; then
+			OSCHOICE="linux"
+		elif [[ "${input}" = "2" ]]; then
+			OSCHOICE="android"
+		else
+			>&2 echo "Invalid OS!"
+			exit 1
+		fi
+
+		echo "Install high quality in-game music? Increases filesize by 194 MB!"
+		echo "[y/n]"
+		read -n1 input
+		echo ""
+		if [[ "${input,,}" = "y" ]]; then
+			HQMUSIC=true
+		fi
+
+		if [ $OSCHOICE = "linux" ]; then
+			echo "Do you want to install AM2R systemwide?"
+			echo "[y/n]"
+			read -n1 input
+			echo ""
+			if [[ "${input,,}" = "y" ]]; then
+				SYSTEMWIDE=true
+			fi
+		fi
+
+		patch_am2r
+		local result=$?
+		echo "Press any key to quit..."
+		read -s -n1 INPUT
+		exit $result
+	fi
+
+	while (( $# > 0 )); do
+		case $1 in
+		-s|--os)
+			OSCHOICE="$2"
+			shift 2 # past argument and value
+			;;
+		-m|--hqmusic)
+			HQMUSIC=true
+			shift # past argument
+			;;
+		-w|--systemwide)
+			SYSTEMWIDE=true
+			shift # past argument
+			;;
+		-a|--appimage)
+			APPIMAGE=true
+			shift # past argument
+			;;
+		-p|--prefix)
+			PREFIX=$(realpath "$2")
+			shift 2 # past argument and value
+			;;
+		-z|--am2rzip)
+			AM2RZIP="$2"
+			shift 2 # past argument and value
+			;;
+		-h|--help)
+			echo -e "-s, --os\t\t\tThe OS to patch to. Valid are \"linux\" and \"android\"."
+			echo -e "-m, --hqmusic\t\t\tIf HighQuality music should be used."
+			echo -e "-w, --systemwide\t\tIf Linux should get installed systemwide."
+			echo -e "-a, --appimage\t\t\tIf an AppImage should get generated."
+			echo -e "-p, --prefix\t\t\tThe prefix used for patching operations. Default for systemwide is \"/usr/local\" and for non-systemwide the folder where this script resides."
+			echo -e "-z, --am2rzip\t\t\tThe path to the AM2R_11 zip or directory. Default is <directory where the script resides>/AM2R_11.zip"
+			exit 0
+			;;
+		*)
+			>&2 echo "Unknown option $1"
+			exit 1
+			;;
+		esac
+	done
+	# restore positional parameters
+	set -- "${POSITIONAL_ARGS[@]}"
+
+	# check if necessary variables are provided
+	if [ -z "$OSCHOICE" ]; then
+		>&2 echo "Missing argument! The OS needs to be provided via the \"--os\" flag! For further info use --help."
```

Comment:
We've already discussed this privately, but I think making the default Linux makes a lot of sense. Most users will want to use that.

Reply:
Done locally.